syntax = "proto3";

option go_package = "./proto";

service VisualService {
    rpc GenerateTimeSeries (TimeSeriesRequest) returns (JsonResponse);
    rpc GenerateMonthlyAverage (DatasetRequest) returns (ImageResponse);
    rpc GenerateVolumeAnalysis (VolumeAnalysisRequest) returns (JsonResponse);
    rpc GeneratePriceAndPercent (PriceAndPercentRequest) returns (JsonResponse);
}

message TimeSeriesRequest {
    string dataset = 1;
    int32 mv = 2;
    string date_range = 3;
}

message VolumeAnalysisRequest {
    string dataset = 1;
    string date_range = 2;
}

message PriceAndPercentRequest {
    string dataset = 1;
    string date_range = 2;
}

message DatasetRequest {
    string dataset = 1;
}

message ImageResponse {
    bytes image_data = 1;
}

message JsonResponse {
    string json_data = 1;
}
